controllers/servers/csi/csi_controller_server.py

```diff
@@ -75,19 +75,30 @@ def CreateVolume(self, request, context):
                     logger.debug("requested size is 0 so the default size will be used : {0} ".format(
                         required_bytes))
 
-                if volume_parameters.virt_snap_func and source_id:
-                    array_mediator.validate_space_efficiency_matches_source(space_efficiency, source_id, source_type)
+                is_virt_snap_func = volume_parameters.virt_snap_func
+                if is_virt_snap_func and source_id:
+                    source_object = array_mediator.get_object_by_id(source_id, source_type, is_virt_snap_func)
+                    if not source_object:
+                        handle_exception("source {}: {} not found".format(source_type, source_id), context,
+                                         grpc.StatusCode.NOT_FOUND, csi_pb2.CreateVolumeResponse)
```

Comment:
don't we need this [here](https://github.com/IBM/ibm-block-csi-driver/blob/58f8bd86ed797c8f74d672e9815f89d0f12627e2/controllers/servers/csi/csi_controller_server.py#L89) as well?

Reply:
if I understand the spec correctly, the source must exist during the creation request handling.
i.e. deleting the source is not supported before finishing the target creation successfully

Reply:
that section is for source of snapshot..

Reply:
source of type "snapshot"? got it

Reply:
done